.github/actions/build-src/action.yaml

```diff
@@ -0,0 +1,69 @@
+name: 'build-src'
+description: 'make source builds for CI'
+inputs:
+  build-hole:
+    description: 'build HOLE2'
+    required: true
+    default: true
+  build-tests:
+    description: 'build MDA tests'
+    required: true
+    default: true
+  build-docs:
+    description: 'build MDA docs'
+    required: true
+    default: false
+
+
+runs:
+  using: "composite"
+  steps:
+    - name: echo_inputs
+      shell: bash -l {0}
+      run: |
+        echo ${{ inputs.build-hole }}
+        echo ${{ inputs.build-tests }}
+        echo ${{ inputs.build-docs }}
+
+    - name: check_setup
+      shell: bash -l {0}
+      run: |
+        # Check OS and python setup prior to builds
+        echo "OS: ${OS_NAME}"
+        which python
+        which pip
+        pip list
+        conda info
+        conda list
+
+    - name: build_hole
+      if: ${{ inputs.build-hole == 'true' }}
+      shell: bash -l {0}
+      run: |
+        # We manually build hole2 to avoid OS incompatibilities
```

Comment:
Separate issue, but isn't this something a hole2 conda package would solve?

Reply:
Yeah - I want to say @orbeckst mentioned looking into that at some point (or was asking for someone to look into that?)

If I remember correctly part of the limitation here is that we wanted to update the upstream repository (since MDAnalysis' version is a fork)? That being said, it's been a year and that PR (https://github.com/osmart/hole2/pull/2) hasn't been merged. I have raised an issue on the MDA's hole2 repo.